### ISSUE TITLE
Improve accessibility for Revision Selector

### ISSUE THREAD

**Issue by mirka** [Automattic/simplenote-electron#980]
The RevisionSelector should act like a modal, in that:

1. It should have role="dialog"
1. It should be `aria-hidden` when not visible on screen
1. <kbd>Esc</kbd> should close it
1. Keyboard focus should jump to the first focusable element when the dialog is activated
1. Keyboard focus should be trapped inside the dialog
1. Keyboard focus should return to the original element when the dialog is closed
